### DISCUSSION TITLE
BGP for Cisco ASA without Loopback Interface

### DISCUSSION THREAD

**Discussion by willimei** [ipspace/netlab#326] [General]
At the moment I am adding the Cisco ASA platform to netsim-tools.
During the implementation of BGP I came across the problem that the ASA does not support loopback interfaces.

What would be the best way to work around this problem? I thought about a config option, which allows to generate the BGP config without a loopback interface and sets different neighbor IPs for every link.

**Comment by ipspace** [comment]
You don't need loopback interfaces for EBGP sessions. Just implement BGP on directly connected interfaces, and I'll add logic to the BGP module that will check whether a device supports IBGP sessions or not.

**Comment by ipspace** [comment]
@willimei implemented vASA (release 1.4.1)
